synapse/handlers/federation.py

```diff
@@ -1498,6 +1543,42 @@ async def _sync_partial_state_room(
                 allow_rejected=True,
             )
             for event in events:
-                await self._federation_event_handler.update_state_for_partial_state_event(
-                    destination, event
-                )
+                for attempt in range(num_destinations):
+                    try:
+                        await self._federation_event_handler.update_state_for_partial_state_event(
+                            destination, event
+                        )
+                        break
+                    except (
+                        InvalidResponseError,
+                        FederationDeniedError,
+                        HttpResponseException,
+                        NotRetryingDestination,
+                        RequestSendFailed,
+                    ) as e:
+                        if attempt == num_destinations - 1:
+                            # We have tried every remote server. Give up.
+                            logger.error(
+                                "Failed to get state for %s at %s from %s because %s. "
+                                "Giving up!",
+                                room_id,
+                                event,
+                                destination,
+                                e,
+                            )
+                            raise
```

Comment:
I'm unsure about what to do if we've tried every destination and none of them worked.